app/controllers/tags_controller.rb

```diff
@@ -129,25 +129,32 @@ def merge
     AuditLog.moderator_audit event_type: 'tag_merge', related: @primary, user: current_user,
                              comment: "#{@subordinate.name} (#{@subordinate.id}) into #{@primary.name} (#{@primary.id})"
 
-    # Take the tag off posts
+    # Replace subordinate with primary, except when a post already has primary (to avoid giving them a duplicate tag)
     posts_sql = 'UPDATE posts INNER JOIN posts_tags ON posts.id = posts_tags.post_id ' \
                 'SET posts.tags_cache = REPLACE(posts.tags_cache, ?, ?) ' \
-                'WHERE posts_tags.tag_id = ?'
-    exec([posts_sql, "\n- #{@subordinate.name}", "\n- #{@primary.name}", @subordinate.id])
+                'WHERE posts_tags.tag_id = ? ' \
+                'AND post_tags.post_id NOT IN (SELECT post_id FROM posts_tags WHERE tag_id = ?)'
```

Comment:
Typo found during testing:

```suggestion
                'AND posts_tags.post_id NOT IN (SELECT post_id FROM posts_tags WHERE tag_id = ?)'
```

Reply:
Applied the change